nucypher/blockchain/eth/agents.py

```diff
@@ -111,7 +111,7 @@ def __repr__(self):
         return r.format(class_name, self.registry, self.registry_contract_name)
 
     def __eq__(self, other):
-        return bool(self.contract_address == other.contract_address)
+        return bool(self.contract.address == other.contract.address)
```

Comment:
Why type casting? Is comparison ever not bool? Or is it a type hint?

Reply:
Just a style habit  I have for `__eq__`.